pkg/authentication/example_test.go

```diff
@@ -69,3 +72,146 @@ func ExampleApiClient_Octocat() {
 	//              ~~~~~~==~==~==~==~~~~~~
 	//                  :~==~==~==~==~~
 }
+
+func Example_ResponseHandlerForRateLimiting() {
+	// this responseHandler function panics:
+	// panic: interface conversion: *http.Response is not serialization.Parsable: missing method GetFieldDeserializers
+	// not sure if i'm mishandling something in this totally non-implemented function or what
```

Comment:
I'm not sure what I'm doing wrong here (possibly related to not implementing any logic at all in the handler), but this panics each call. More investigation is required.